studymanager/src/main/java/io/redlink/more/studymanager/properties/GatewayProperties.java

```diff
@@ -8,18 +8,21 @@
  */
 package io.redlink.more.studymanager.properties;
 
+import io.redlink.more.studymanager.model.Participant;
 import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.web.util.UriComponentsBuilder;
 
-@ConfigurationProperties(prefix = "more.gateway")
-public class GatewayProperties {
-    private String baseUrl;
+import java.net.URI;
 
-    public String getBaseUrl() {
-        return baseUrl;
-    }
+@ConfigurationProperties(prefix = "more.gateway")
+public record GatewayProperties(String baseUrl) {
+    public URI generateSignupUrl(Participant participant) {
+        if (participant.getRegistrationToken() == null) return null;
 
-    public GatewayProperties setBaseUrl(String baseUrl) {
-        this.baseUrl = baseUrl;
-        return this;
+        return UriComponentsBuilder.fromUriString(baseUrl)
+                .path("signup")
+                .queryParam("token", participant.getRegistrationToken())
+                .build()
```

Comment:
Be careful here that the registration-token is properly url-encoded.

`UriComponenteBuilder` provides some support for this:
```suggestion
                .queryParam("token", "{token}")
                .build("token", participant.getRegistrationToken())
```

Reply:
Done in https://github.com/MORE-Platform/more-studymanager-backend/pull/361/commits/68f46b2b29ca2cd9111cad67bed8c2b6428b4e5d